src/WebJobs.Extensions.NotificationHubs/Bindings/NotificationHubAsyncCollector.cs

```diff
@@ -4,23 +4,44 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.NotificationHubs;
+using Microsoft.Azure.WebJobs.Host;
 
 namespace Microsoft.Azure.WebJobs.Extensions.NotificationHubs
 {
     internal class NotificationHubAsyncCollector : IAsyncCollector<Notification>
     {
         private INotificationHubClientService _notificationHubclientService;
         private string _tagExpression;
+        private bool _enableTestSend;
+        private TraceWriter _traceWriter;
 
-        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression)
+        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression, bool enableTestSend, TraceWriter traceWriter)
         {
             _notificationHubclientService = clientService;
             _tagExpression = tagExpression;
+            _enableTestSend = enableTestSend;
+            _traceWriter = traceWriter;
         }
 
         public async Task AddAsync(Notification item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            NotificationOutcome notificationOutcome = await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            if (_enableTestSend)
+            {
+                _traceWriter.Info($"[NotificationHubsDebugLog] TrackingId = {notificationOutcome.TrackingId}");
```

Comment:
@pragnagopa -- It's not a huge deal, but I still think you should build up a string with these `if` clauses and write out once at the end of the method. Otherwise, you may not be able to link the TrackingId to the log below it (it's possible for other logs to sneak in between).